neural_net_basics/neural-network.md

```diff
@@ -0,0 +1,137 @@
+## Intro
+Recently I've been learning about Neural Networks and found myself wanting to write about the how they work from a basic level.
+
+## What is a neural network made of?
+
+All neural networks are made up of a network of neurons, much like the brain that they model, but what is a neuron?
+
+A neuron is made up of it's inputs, some weights, a bias and activation function and an output and was originally intended to replicate the biological neuron.
+
+There is a one to one ratio of inputs to weights, which when combined gives a series of weighted inputs that get summed together. To this sum a bias get's added.
+
+This can be represented mathematically as follows
+
+$$ f(x)= \sum^n_{i=1}w_{i}x_{i}+b $$
+
+* $w$ is the weight
+* $x$ is the input
+* $b$ is the bias
+* $n$ is the number of inputs
+
+
+![png](/neural_net_basics//assets//neuron.png)
+
+## Why is an activation function important?
+A neuron, with a single input, is a linear function, $y = ax+b$, and as such can only form simple decision boundaries. An increase in the number of inputs and neurons in the neural network will still result in a linear function. This is because multiple combined linear functions always results in a linear function. 
+
+To resolve this issue, and allow the network to find complex decision boundaries, one that wiggles around separating multiple classes, we need to use activation functions, they add non-linearity.
+
+### Step Function
+![png](/neural_net_basics//assets//step.png) 
+$$ f(x) =\begin{cases}0 & x < 0\\1 & x  \geq  0\end{cases} $$
+
+The original activation function, used in the original perceptron in the 1950's. This function converts the output of the summation function into a binary value.
+
+Due to the nature of the step in the function, changes to the weights and bias make no difference until the output crosses the step boundary, at which point it flips the binary value. Due to a very small change to the weight having such a dramatic effect, it is hard to train the neural network.
+
+### Logistic Function a.k.a Sigmoid function
+![png](/neural_net_basics//assets//sigmoid.png)
+$$ f(x) = \frac{1}{1 + e^{-x}}$$
+The Logistic function, a variant of the sigmoid function, initially replaced the step function. Like the Step function it limits the output of a neuron between 0 and 1, but doesn't have the same issues around the zero mark. This makes it easier to train as a change to the weights and biases will always have an effect, except at the extremes.
+
+### Rectified Linear Unit (ReLU)
+![png](/neural_net_basics//assets//relu.png)
+$$ f(x) = max(0,x) $$
+The Rectified Linear  (ReLU) has become the most popular and in most cases the default activation function to use for deep neural networks. It takes the maximum value of either 0 or the input, essentially putting any negative input value to 0 immediately.
+
+### SoftMax
+$$ \delta(z) = \frac{e^{z_{i}}}{\sum_{j=1}^{K}e^{z_{j}}} $$
```

Comment:
It might be worth pointing out that SoftMax is dissimilar to the others in that it takes a vector